src/materials/ConcreteASREigenstrain.C

```diff
@@ -0,0 +1,278 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteASREigenstrain.h"
+
+registerMooseObject("BlackBearApp", ConcreteASREigenstrain);
+
+template <>
+InputParameters
+validParams<ConcreteASREigenstrain>()
+{
+  InputParameters params = validParams<ConcreteExpansionEigenstrainBase>();
+
+  MooseEnum expansion_type("Isotropic Anisotropic");
+  params.addParam<MooseEnum>(
+      "expansion", expansion_type, "Type of expansion resulting from volumetric strain");
+  params.addRequiredCoupledVar("temperature", "Coupled temperature");
+  params.addRequiredCoupledVar("relative_humidity", "Coupled relative humidity");
+
+  params.addParam<Real>(
+      "rh_exponent",
+      0.0,
+      "Power to which relative humidity is raised in computation of ASR volumetric strain");
+
+  params.addParam<Real>(
+      "ASR_vol_expansion",
+      0.005,
```

Comment:
I would like to have no default value here if possible given the quite high variability in concrete reactivity. Idem for tau_c_T0 and tau_L_T0.